### ISSUE TITLE
LDAP group sync broken on 0.34.0

### ISSUE THREAD

**Issue by gvisniuc** [metabase/metabase#11617]
**Describe the bug**
After upgrading from **0.33.6** to **0.34.0** whenever users logged out and logged back in they would not be placed in the appropriate mapped group in the LDAP settings. Instead they are placed in the **Default** group and loose all permissions.

If permissions are set manually they persist until the user logs out and logs back in.

I have checked the deployment logs to reproduce the issue but I have not found anything significant. I have now reverted to **0.33.6** and is working fine again.

I have tried a different upgrade path as well with no success.

**To Reproduce**
1. Upgrade to **0.33.4**
2. Log out
3. Log in
4. Notice from the admin panel that user is now in the **Default** group
5. Revert to **0.33.6** and regain permissions

**Expected behavior**
Users would be placed in the appropriate LDAP group mapping.

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

```{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.3+7",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.3",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.3+7",
    "os.name": "Linux",
    "os.version": "4.19.0-6-cloud-amd64",
    "user.language": "en",
    "user.timezone": "Europe/Amsterdam"
  },
  "metabase-info": {
    "databases": [
      "redshift",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2019-11-19",
      "tag": "v0.33.6",
      "branch": "release-0.33.x",
      "hash": "be1e0e1"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

**Severity**
Blocking users from all resources if they logout or if they need to login.
Their permissions need to be manually set, but are reset if they logout.

**Comment by mazameli** [comment]
Thanks for reporting this, and sorry for the troubles, George. We'll be looking into this.

**Comment by gvisniuc** [comment]
No troubles at all @mazameli, thanks for looking into it!

**Comment by walterl** [comment]
Hi @gvisniuc. I'm currently investigating this issue. You wouldn't happen to be authenticating against an Active Directory server, would you?

**Comment by gvisniuc** [comment]
Hi @walterl, we are using FreeIPA for our LDAP installation. Let me know if I can provide more info or if can help somehow.

**Comment by walterl** [comment]
@gvisniuc I found a possible bug that is specific to the way that AD communicates groups, and related to the only change in the LDAP code between the versions you mentioned. So I thought (hoped) that it would be that.

In the fix for #11411, all returned LDAP keys were lower cased (to allow case insensitive lookups), but the AD-specific [`memberOf` key](https://github.com/metabase/metabase/blob/master/src/metabase/integrations/ldap.clj#L202) for pulling out groups wasn't similarly lower cased.

While I try to stand up a FreeIPA server for testing, do you know if FreeIPA provides group info in a `memberOf` overlay, or did you have to configure the _Search base for groups_ setting in Metabase?

**Comment by gvisniuc** [comment]
@walterl In the case of FreeIPA it provides the group information via the `memberOf` overlay. I did not have to configure the `Search base` field in Metabase (or any other apps that we use).

**Comment by walterl** [comment]
Then it's likely the same issue that's affecting you. Are you perhaps in a position where you can test if this patch fixes the issue for you?

```diff
diff --git a/src/metabase/integrations/ldap.clj b/src/metabase/integrations/ldap.clj
index 7705366c1..536086b9e 100644
--- a/src/metabase/integrations/ldap.clj
+++ b/src/metabase/integrations/ldap.clj
@@ -199,7 +199,7 @@
           :groups     (when (ldap-group-sync)
                         ;; ActiveDirectory (and others?) will supply a `memberOf` overlay attribute for groups
                         ;; Otherwise we have to make the inverse query to get them
-                        (or (:memberOf result) (get-user-groups dn) []))})))))
+                        (or (:memberof result) (get-user-groups dn) []))})))))
 
 (defn verify-password
   "Verifies if the supplied password is valid for the `user-info` (from `find-user`) or DN."
```

**Comment by gvisniuc** [comment]
@walterl Awesome. 

Yes, will I need to rebuild the project based on your branch? It would be great if you could provide a Docker image (or the jar file?). In either case let me know how you'd like to proceed.

**Comment by walterl** [comment]
You got me to have CircleCI build us (you) a [jar file]https://193924-30203935-gh.circle-artifacts.com/0/home/circleci/metabase/metabase/target/uberjar/metabase.jar) :slightly_smiling_face: 

Could you please see if that fixes the issue for you?

**Comment by gvisniuc** [comment]
@walterl Seems to work great. These are the steps I took to replicate:
1) Added jar file inside the container and started it
2) Added Metabase  LDAP auth settings
3) Added dummy group
4) Added user to dummy group in FreeIPA
5) Created mapping in Metabase LDAP settings
6) Logged out and logged back in, user is placed in the appropriate group

**Comment by walterl** [comment]
Great! Thanks for your help. I'll get that PR reviewed, and included in the 0.34.1 release.

**Comment by selfing12** [comment]
@gvisniuc Good afternoon!Could you please share the settings of your metaybase for working with ldap freeipa, authorization does not work for me and I cannot find a solution.Error: did not match the stored password.No matter how I try to set it up.
